bulk_photo_nodes.module

```diff
@@ -636,9 +636,16 @@ function bulk_photo_nodes_save_node($node_type, $node_fields, $node_overrides, &
   if ($options['directory'] == '/') {
     $options['directory'] = '';
   }
-  $destination = $options['scheme'] . $image_field->uid . '-' . $images_count . '.' . $extension;
-  $image_field->filename = $image_field->uid . '-' . $images_count . '.' . $extension;
-  $image_field_ref = (array) file_move($image_field, $destination, FILE_EXISTS_REPLACE);
+
+  // Allow to use the same filename, but update image information
+  // ONLY if image URI has changed.
+  $destination = $options['scheme'] . $image_field->filename;
+  if ($destination == $image_field->uri) {
+    $image_field_ref = $node->{$image_field_name}[LANGUAGE_NONE][0];
```

Comment:
You already set $image_field_ref on line 627. so you aren't repeating yoruself why not switch your logic on your if statement making it

```
if ($destination != $image_field->uri) {
  $image_field_ref = (array) file_move...;
}
```